### ISSUE TITLE
Please help!!

### ISSUE THREAD

**Issue by lollerbay** [eforce67/BrawlStars-ComputerVision#2]
This is the error I am getting after executing "train.py"
using LD player 9, all requirements installed, using python 3.10.11

beginning the match
None
Exception in thread Thread-17 (run_simulation):
Traceback (most recent call last):
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\Administrator\Downloads\BrawlStars-ComputerVision-main\train.py", line 298, in run_simulation
    prediction = load_image_and_detect(f'screen_{emulator_name}.png')
  File "C:\Users\Administrator\Downloads\BrawlStars-ComputerVision-main\train.py", line 62, in load_image_and_detect
    net = YOLO(model=model)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\models\yolo\model.py", line 23, in __init__
    super().__init__(model=model, task=task, verbose=verbose)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\engine\model.py", line 152, in __init__
    self._load(model, task=task)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\engine\model.py", line 241, in _load
    self.model, self.ckpt = attempt_load_one_weight(weights)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\nn\tasks.py", line 806, in attempt_load_one_weight
    ckpt, weight = torch_safe_load(weight)  # load ckpt
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\nn\tasks.py", line 732, in torch_safe_load
    ckpt = torch.load(file, map_location="cpu")
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\serialization.py", line 791, in load
    with _open_file_like(f, 'rb') as opened_file:
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\serialization.py", line 271, in _open_file_like
    return _open_file(name_or_buffer, mode)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\serialization.py", line 252, in __init__
    super().__init__(open(name, mode))
FileNotFoundError: [Errno 2] No such file or directory: 'resources\\models\\best2.pt'
finished threading

**Comment by eforce67** [comment]
It looks like the script is not being able to find your LDPlayer window. Check your configuration and the default name for your main emulator which should be LDPlayer. If you have other emulators, please add their name to the config and start the emulator before you run the train.py script

Be sure to also download graphviz here since it does not seem to be working normally for you https://graphviz.org/download/

Let me know how it goes!

**Comment by lollerbay** [comment]
Hey thank you so much for replying, unfortunately that's the error I am getting on repeat after starting a match even after installing graphviz and changing the emulator name in settings.yaml (LDplayer). It does find my LDplayer window, starts a match and then gives me this error, might be ultralytics error? Should I use bluestacks? or perhaphs another python version? It seems like the problem is originating from my python installation especially ultralytics and pytorch In the last line of error "resources\\models\\best2.pt' should the best2.pt model be already present before training?

**Comment by eforce67** [comment]
As I looked more clearly at the problem now, please make sure to download a model checkpoint from: https://mega.nz/folder/uGogBbZB#hWY8tXO0kOGGrGU5vGityw/file/qW51zZaI

And name best.pt to best2.pt and make sure to place it in resources/models/

The model checkpoint is not originally presented in the folder because the model size is too large for github.

**Comment by lollerbay** [comment]
Hey thanks its working now. It's moving the brawler (getting stuck on walls alot) but not shooting anyone and also it doesnt know when the match ends to reque so I have do it manually, any brawler you suggest in particular? maybe it's the training in process, I'll check on it after maybe 10-20 matches. BTW how to stop the training process as intentionally because everytime I close the comand prompt it havent generated any files. Is there any way to visualize what the Yolo is seeing? Because I am running it on CPU (Ryzen 5 3550h) As I don't have a Nvida GPU so I am worried it might be running too slow to process and react quick enough. Thank you so much for for helping me out ❤️

Edit: hey I tried about 15 matches and maybe something is wrong it's not shooting at all not following teammates, running into walls and getting stuck into it forever, running towards enemies straight, Idk if only 15 matches should have showed any improvements but will  100-150 runs be enough to get it running better?

Edit2: Its capturing only half of the screen, I think this might be the culprit, I have attached the screen_LDplayer.png while it was capturing frames.
![screen_LDPlayer](https://github.com/eforce67/BrawlStars-ComputerVision/assets/81859171/b19ba5a9-316c-4964-b199-d8fefb572c79)


https://github.com/eforce67/BrawlStars-ComputerVision/assets/81859171/e3317da8-191f-4666-a385-99dfa3ef1aaf


Maybe my PC is not strong enough for YOLO, Thanks you for helping and making this project.
